### ISSUE TITLE
Status check

### ISSUE THREAD

**Issue by DougAnderson444** [Defluencer/rust-defluencer#17]
Hi @SionoiS,

I've been looking for a Prolly Tree impl in Rust and found your library through various IPLD / filecoin searches and whatnot.

I am just wondering how far along you are or if you want/need any help in any area.

I have a bunch of key-values that I want to store then index (and search) across my own nodes and other peoples' nodes and I'm trying to see how Defluencer can help with that.

Wondering where you are at with this project!

**Comment by SionoiS** [comment]
Hello! @DougAnderson444 

The Prolly tree impl. is fully functional and tested. The MST one not so much, there's a couple of edge cases that need ironing out.

The problem is the lack of standard Traits for IPFS API. It should be easy to substitute any other IPFS impl. but some elbow grease required.

**Comment by DougAnderson444** [comment]
Yeah I noticed the ipfs is this struct `use ipfs_api::{responses::Codec, IpfsService};`  instead of a trait.

**Comment by SionoiS** [comment]
> Yeah I noticed the ipfs is this struct `use ipfs_api::{responses::Codec, IpfsService};` instead of a trait.

The best would be to have a some kind of standard but rust is low priority for IPFS folks so :shrug: 

I was thinking of something like
```
trait FullAPI: GetAPI, BlockAPI, etc..
```
so that you can follow the standard but not forced to implement everything. WDYT?
